src/rngs/adapter/reseeding.rs

```diff
@@ -193,11 +207,13 @@ where R: BlockRngCore + SeedableRng,
                     error from source: {}", delay, e);
             delay
         } else {
-            self.threshold
+            let num_bytes =
+                results.as_ref().len() * size_of::<<R as BlockRngCore>::Item>();
+            self.fork_counter = fork_counter;
+            self.threshold - num_bytes as i64
```

Comment:
`num_bytes` should be small relative to `threshold` and we aren't exact with its handling anyway, so is there any point subtracting it? You've already adjusted the `should_retry` behaviour here.

Reply:
Actually, why make this change at all (i.e. why not revert the `num_bytes` subtraction to how it was)?

Reply:
Forgot (and rediscovered) why I changed things here, the old core was slightly confusing for me.
`threshold` is set to `0` on `ErrorKind::Transient`. Then `bytes_until_reseed` was `threshold - num_bytes`. Does the `Transient` work correctly?

I have partly undone the changes, and only set the delay caused by `ErrorKind::Transient` to `num_bytes`, to get the right value when logging.